### PR TITLE
feat(EditableList): Add editable list widget

### PR DESCRIPTION
**Pull request by floryst**
Adds initial EditableListWidget. Supports adding, deleting, and resorting.

image

### CODE REVIEW

File: src/React/Widgets/EditableListWidget/EditableList.mcss [hunk 1]
```diff
@@ -0,0 +1,90 @@
+.list {
```
Comment (jourdain) [inline src/React/Widgets/EditableListWidget/EditableList.mcss]: The style files, need to go underneath 'style'...

**Comment by jourdain** [comment]
Can you add a prop (optional) for which we can give a sort function when we want to auto sort the list...

File: src/React/Widgets/EditableListWidget/EditableList.mcss (hunk 1 above)
Comment (jourdain) [reply src/React/Widgets/EditableListWidget/EditableList.mcss]: This allow restyling when used in other project.

**Comment by floryst** [comment]
Do you mean auto-sorting the list when adding a new element, or adding a button to sort the list based on some supplied sorting function?

**Comment by jourdain** [comment]
Auto sorting like we have with the radius for the cells.

**Comment by floryst** [comment]
Do you mean auto-sorting the list when adding a new element, or adding a button to sort the list based on some supplied sorting function?

**Comment by floryst** [comment]
Right now the EditableListWidget doesn't actually manage data sorting order. If order changes due to drag-n-drop, then `onSortChange` is invoked with `srcIndex` and `dstIndex`, but the widget itself doesn't actually change data order. I think that should be managed by whoever is passing in the data.

I could visually auto-sort the rows, but it won't change the underlying data order. Is that what you prefer?

**Comment by jourdain** [comment]
I guess that should be the outside widget that will do the sorting like you were suggesting. You should be good on that end.

**Comment by floryst** [comment]
Moved the styles to the proper directory.

**Pull request merged**